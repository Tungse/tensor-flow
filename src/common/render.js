export const renderMedia = (item) => {
  switch (item['@type']) {
    case 'ImageObject':
      return `<img class="" src="${item.contentUrl}" alt="">`
    case 'VideoObject':
      return `<iframe class="" src="${item.embedUrl}"></iframe>`
    default:
      return ``
  }
}
