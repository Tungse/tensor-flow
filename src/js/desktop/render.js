import Filer from '../../../node_modules/filer-js-sdk/dist/filer.js'

/**
 * Renders Stage depending on current state
 * @return {string} template string
 */
export const renderStage = (state, settings) => {
  const prevPage = state.data.itemListElement[state.currentPage - 2]
  const page = state.data.itemListElement[state.currentPage - 1]
  const nextPage = state.data.itemListElement[state.currentPage]

  // on last page (endcard), we dont show stage
  if (state.currentPage === state.length) {
    return ''
  }

  return `
    <div class="smb-gallery-stage smb-gallery-desktop">
      <h2>${page.item.headline}</h2>
      <div class="smb-gallery-header">
        ${state.currentPage > 1 ? `
          <a role="smb-gallery-prev" class="smb-gallery-nav smb-gallery-nav-left" href="${prevPage.item.url}">
            <div class="smb-gallery-button">
              ${settings.prevIcon}
            </div>
          </a>
        ` : ''}
        <div class="smb-gallery-media ${page.item['@type']}">
          ${renderMedia(page.item)}
        </div>
        ${state.currentPage < state.length ? `
          <a role="smb-gallery-next" class="smb-gallery-nav smb-gallery-nav-right" href="${nextPage.item.url}">
            <div class="smb-gallery-button">
              ${settings.nextIcon}
            </div>
          </a>
        ` : ''}
        <div class="smb-gallery-info">
          ${page.item.copyrightHolder ? `
          <small>Bildquelle: ${page.item.copyrightHolder}</small>
          ` : '<small></small>'}
          <small>${state.currentPage} / ${state.length}</small>
        </div>
      </div>

    </div>
  `
}

/**
 * Renders MediaItem
 * @return {string} template string
 */
const renderMedia = (item) => {
  switch (item['@type']) {
    case 'ImageObject':
      return `
        <div class="embed-responsive" style="padding-bottom: 450px">
          <img class="embed-extended embed-responsive-item" src="${Filer.createVariantUrl(item.contentUrl, [['rcm', 0, 450, 'u']])}" alt="">
        </div>
      `
    case 'VideoObject':
      return `
        ${item.embedUrl.includes('youtube') || item.embedUrl.includes('youtu.be') ? `
          <div class="embed-responsive embed-responsive-16by9">
            <div class="embed-responsive-item" data-embedo-url="${item.embedUrl}"></div>
          </div>
        ` : `
          <div class="embed-responsive embed-responsive-16by9">
            <iframe allowfullscreen class="embed-responsive-item" src="${item.embedUrl}"></iframe>
          </div>
        `}
      `
    case 'SocialMediaPosting':
      return `<div data-embedo-url="${item.sharedContent.url}"></div>`
    default:
      return ``
  }
}

/**
 * Renders Content depending on current state
 * @return {string} template string
 */
export const renderContent = (state, settings) => {
  const page = state.data.itemListElement[state.currentPage - 1]

  return `
    <div class="smg-gallery-body smb-gallery-desktop">
      ${page.item.description}

      <div class="smb-gallery-btn-nav">
        <div>
        ${state.referrer ? `
          <a role="smb-gallery-back" class="btn btn-link smb-gallery-back" href="${state.referrer}"><i class="fas fa-angle-left"></i> zurück zum Artikel</a>
        ` : ''}
        </div>
        <div>
          <a role="smb-gallery-prev" class="btn btn-primary ${state.currentPage === 1 ? 'disabled' : ''}" href="#"><i class="fas fa-angle-left"></i> zurück</a>
          <a role="smb-gallery-next" class="btn btn-primary ${state.currentPage === state.length ? 'disabled' : ''}" href="#">weiter <i class="fas fa-angle-right"></i></a>
        </div>
      </div>
    </div>
  `
}
