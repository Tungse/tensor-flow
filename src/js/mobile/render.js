import Filer from 'filer-js-sdk'

/**
 * Build template string for gallery
 * @return {string} template string
 */
const renderGalleryItems = (state) => {
  return `
      <div class="smb-gallery-mobile">
        ${state.referrer ? `
          <a role="smb-gallery-back" class="btn btn-link smb-gallery-back" href="${state.referrer}"><i class="fas fa-angle-left"></i> zur√ºck zum Artikel</a>
        ` : ''}
      ${state.data.itemListElement.map((page, i) => `
        <div class="smb-gallery-item">
          <h2>${page.item.headline}</h2>
          <div class="smb-gallery-media ${page.item['@type']}">
            ${renderMedia(page.item)}
              <div class="smb-gallery-info">
              ${page.item.copyrightHolder ? `
                <small>Bildquelle: ${page.item.copyrightHolder}</small>
              ` : '<small></small>'}
                <small>${i + 1} / ${state.length}</small>
              </div>
          </div>
          <div class="smb-gallery-content">
            ${page.item.description}
          </div>
          <div class="smb-gallery-ed-container">
            <div data-slotname="${getSlotName(i)}"></div>
          </div>
        </div>
    `.trim()).join('')}
    </div>
  `
}

/**
 * Build templates-strings for different media items
 * @return {string}
 */
const renderMedia = (item) => {
  switch (item['@type']) {
    case 'ImageObject':
      return `
        ${item.width > 0 && item.height > 0 ? `
          <div class="embed-responsive" style="padding-bottom: ${item.height / item.width * 100}%">
            <img class="embed-responsive-item lazyload" data-src="${Filer.createVariantUrl(item.contentUrl, [['rcm', 480, 0, 'u']])}" alt="">
          </div>
        ` : `
          <img class="lazyload" data-src="${Filer.createVariantUrl(item.contentUrl, [['rcm', 480, 0, 'u']])}" alt="">
        `}
      `
    case 'VideoObject':
      return `
        ${item.embedUrl.includes('youtube') || item.embedUrl.includes('youtu.be') ? `
          <div class="embed-responsive embed-responsive-16by9">
            <div class="embed-responsive-item" data-role="embedo" data-url="${item.embedUrl}"></div>
          </div>
        ` : `
          <div class="embed-responsive embed-responsive-16by9">
            <iframe allowfullscreen class="embed-responsive-item lazyload" data-src="${item.embedUrl}"></iframe>
          </div>
        `}
      `

    case 'SocialMediaPosting':
      return `<div data-role="embedo" data-url="${item.sharedContent.url}"></div>`
  }
}

/**
 * Find slotname by index
 * @param  {Int} i
 * @return {String}
 */
const getSlotName = (i) => {
  switch (i % 4) {
    case 0:
      return 'galleryad'
    case 1:
      return 'galleryad2'
    case 2:
      return 'galleryad3'
    case 3:
      return 'galleryad4'
  }
}

export default renderGalleryItems
