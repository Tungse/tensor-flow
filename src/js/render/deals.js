import store from '../store/store.js'

const renderDeals = () => {
  const deals = document.querySelector('[data-role="smb-phone-plan-deals"]')

  deals.innerHTML = `
    <ul class="list-group">
      ${store.get().deals.map((deal, index) => `
        <li class="list-group-item">
          <div class="listing-text">
            <h4>${deal.title}</h4>
            <p>${deal.id}</p>
          </div>
          <div class="listing-price">${deal.price}</div>
          <div class="clearfix">
            <div class="better-options pull-left">
              ${deal.options.map((option, index) => `
                <span>${option}</span>
              `.trim()).join('')}
            </div>
            <button class="btn btn-primary pull-right">Zum Angebot</button>
          </div>
        </li>
      `.trim()).join('')}
    </ul>
  `
}

export default renderDeals
