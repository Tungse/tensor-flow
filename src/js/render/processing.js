import store from '../store/store.js'

/**
 * return random facts
 * @returns {string}
 */
const getRandomFacts = () => {
  const facts = [
    '57 Millionen Deutsche besitzen ein Smartphone. (Quelle: statista)',
    '43% der 20 bis 29-Jährigen in Deutschland nutzen ihr Smartphone mehr als 120 Minuten täglich. (Quelle: statista)',
    'Die Netzabdeckung mit 4G liegt in Deutschland bei 65%, in Norwegen bei 92%. (Quelle: statista)',
  ]

  return facts[Math.floor(Math.random() * facts.length)]
}

/**
 * render processing with a random fact
 */
export const renderProcessing = () => {
  const container = store.get().container.querySelector('[data-role="smb-phone-plan-processing"]')

  if (container) {
    container.innerHTML = `
      <hr />
      <div class="smb-phone-plan-loading-dot-container text-center">
        <p>
          <span class="smb-phone-plan-loading-dot"></span>
          <span class="smb-phone-plan-loading-dot"></span>
          <span class="smb-phone-plan-loading-dot"></span>
        </p>
      </div>
      <p class="text-center">${getRandomFacts()}</p>
    `
  }
}

/**
 * remove processing
 */
export const removeProcessing = () => {
  const processing = store.get().container.querySelector('[data-role="smb-phone-plan-processing"]')

  if (processing) {
    processing.remove()
  }
}
