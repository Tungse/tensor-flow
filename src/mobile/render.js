import Filer from 'filer-js-sdk'

/**
 * Build template string for gallery
 * @return {string} template string
 */
const renderGalleryItems = (state) => {
  return `
      <div class="smb-gallery-mobile">
      ${state.data.itemListElement.map((page, i) => `
        <div class="smb-gallery-item">
          <div class="smb-gallery-media ${page.item['@type']}">
            ${renderMedia(page.item)}
          </div>
          ${page.item.copyrightHolder ? `
          <div class="smb-gallery-info">
            <small>Bildquelle: ${page.item.copyrightHolder}</small>
          </div>
          ` : ''}
          <div class="smb-gallery-content">
          ${i === 0 ? `
            <h1>${page.item.headline}</h1>
          ` : `
            <h2>${page.item.headline}</h2>
          `}
            ${page.item.description}
          </div>
          <div class="smb-gallery-ed-container">
            <div data-slotname="${getSlotName(i)}"></div>
          </div>
        </div>
    `.trim()).join('')}
    </div>
  `
}

/**
 * Build templates-strings for different media items
 * @return {string}
 */
const renderMedia = (item) => {
  switch (item['@type']) {
    case 'ImageObject':
      return `
        ${item.width > 0 && item.height > 0 ? `
          <div class="embed-responsive" style="padding-bottom: ${item.height / item.width * 100}%">
            <img class="embed-responsive-item lazyload" data-src="${Filer.createVariantUrl(item.contentUrl, [['rcm', 480, 0, 'u']])}" alt="">
          </div>
        ` : `
          <img class="lazyload" data-src="${Filer.createVariantUrl(item.contentUrl, [['rcm', 480, 0, 'u']])}" alt="">
        `}
      `
    case 'VideoObject':
      return `<iframe class="lazyload" data-src="${item.embedUrl}"></iframe>`
    case 'SocialMediaPosting':
      return `<div data-role="embedo" data-url="${item.sharedContent.url}"></div>`
  }
}

/**
 * Find slotname by index
 * @param  {Int} i
 * @return {String}
 */
const getSlotName = (i) => {
  switch (i % 4) {
    case 0:
      return 'galleryad'
    case 1:
      return 'galleryad2'
    case 2:
      return 'galleryad3'
    case 3:
      return 'galleryad4'
  }
}

export default renderGalleryItems
