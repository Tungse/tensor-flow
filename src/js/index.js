import '../stylesheets/gallery.scss'
import desktop from './desktop/init.js'
import mobile from './mobile/init.js'

if (process.env.NODE_ENV === 'development') {
  import('../stylesheets/demo.scss')
}

const init = (options) => {
  if (options && options.device === 'smartphone') {
    mobile(options)
  } else {
    desktop(options)
  }
}

export default init
