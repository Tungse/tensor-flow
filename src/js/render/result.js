import store from '../store/store.js'
import * as track from '../common/tracking.js'
import { validateEmail } from '../common/validation.js'

/**
 * render user result
 */
const renderResult = () => {
  const content = getContentByResult()
  const result = document.querySelector('[data-role="smb-phone-plan-result"]')

  if (result) {
    result.innerHTML = `
      <hr />
      <h3 class="">Dein Ergebnis</h3>
      <div class="smb-phone-plan-progress">
        <div class="smb-phone-plan-progress-bar ${content.progressClass}" data-role="smb-phone-plan-progress-bar"></div>
      </div>
      ${store.get().resultCategory < 2 ? `
        <strong>Dein Tarif ist ${content.categoryText}. Du zahlst ${store.get().priceDiffence}€ zu viel im Monat. Hier gibt es bessere Alternativen:</strong>
      ` : `
        <strong>Herzlichen Glückwünsch. Du hast einen tollen Tarif.</strong>
        ${store.get().emailSended === false ? `
          <hr />
          <div data-role="smb-phone-plan-email">
            <p>Schick mir eine E-Mail wenn es ein besseres Angebot gibt.</p>
            <div class="row">
              <div class="form-group col-sm-6">
                <div class="input-group">
                  <div class="input-group-addon">@</div>
                  <input type="email" class="form-control" data-role="smb-phone-plan-email-input" placeholder="E-Mail">
                </div>
              </div>
              <div class="form-group col-sm-6">
                <button class="btn btn-primary btn-block" data-role="smb-phone-plan-send-email">Absenden</button>
              </div>
            </div>
          </div>
        ` : ``}
      `}
    `

    setProgressBarTransition(store.get().resultPercent)
    listenToSendEmailClick()
    track.result(content.categoryText)
  }
}

/**
 * return result content by user's result category
 * @returns {{progressClass: string, categoryText: string}}
 */
const getContentByResult = () => {
  let content = {
    categoryText: 'schlecht',
    progressClass: 'bg-danger',
  }

  if (store.get().resultCategory === 1) {
    content.categoryText = 'ok'
    content.progressClass = 'bg-warning'
  }

  if (store.get().resultCategory === 2) {
    content.categoryText = 'gut'
    content.progressClass = 'bg-success'
  }

  return content
}

/**
 * workaround to animate progress bar
 * @param progressWidth
 */
const setProgressBarTransition = (progressWidth) => {
  const progressBar = document.querySelector('[data-role="smb-phone-plan-progress-bar"]')

  setTimeout(() => {
    progressBar.style.width = `${progressWidth}%`
  }, 100)
}

/**
 * event listerner for send email button
 */
const listenToSendEmailClick = () => {
  const sendButton = document.querySelector('[data-role="smb-phone-plan-send-email"]')

  if (sendButton === null) {
    return
  }

  sendButton.addEventListener('click', () => {
    const emailForm = document.querySelector('[data-role="smb-phone-plan-email"]')
    const email = document.querySelector('[data-role="smb-phone-plan-email-input"]')

    if (emailForm === null || email === null) {
      return
    }

    if (validateEmail(email.value) === false) {
      email.parentElement.classList.add('has-error')
      return
    }

    emailForm.innerHTML = `<p>Vielen Dank.</p>`
    store.set({ emailSended: true })
    track.sendEmail()
  })
}

export default renderResult
