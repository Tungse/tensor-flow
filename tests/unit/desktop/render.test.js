import { renderStage, renderContent } from '../../../src/js/desktop/render.js'

test('Gallery markup is rendered as expected', () => {
  const state = {
    data: {
      itemListElement: [
        {
          '@type': 'ListItem',
          'position': 1,
          'item': {
            '@type': 'ImageObject',
            'caption': '',
            'contentUrl': '//static.kino.de/wp-content/uploads/2018/05/geplante-star-wars-filme-7.jpg',
            'copyrightHolder': 'Disney',
            'copyrightYear': '2018',
            'datePublished': '2018-10-05T14:28:22',
            'width': 3500,
            'height': 1750,
            'description': 'Description 1',
            'headline': 'Headline 1',
            'url': 'https://www.example.de',
          },
        },
      ],
    },
    currentPage: 1,
  }

  const stage = renderStage(state)
  expect(stage).toMatchSnapshot()

  const content = renderContent(state)
  expect(content).toMatchSnapshot()
})
