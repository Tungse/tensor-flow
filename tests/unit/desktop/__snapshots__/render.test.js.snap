// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Gallery markup is rendered as expected 1`] = `
"
    <div class=\\"smb-gallery-stage smb-gallery-desktop\\">
      <h2>Headline 1</h2>
      <div class=\\"smb-gallery-header\\">
        
        <div class=\\"smb-gallery-media ImageObject\\">
          
        <div class=\\"embed-responsive\\" style=\\"padding-bottom: 450px\\">
          <img class=\\"embed-extended embed-responsive-item\\" src=\\"//static.kino.de/wp-content/uploads/2018/05/geplante-star-wars-filme-7-rcm0x450u.jpg\\" alt=\\"\\">
        </div>
      
        </div>
        
        <div class=\\"smb-gallery-info\\">
          
          <small>Bildquelle: Disney</small>
          
          <small>1 / undefined</small>
        </div>
      </div>

    </div>
  "
`;

exports[`Gallery markup is rendered as expected 2`] = `
"
    <div class=\\"smg-gallery-body smb-gallery-desktop\\">
      Description 1

      <div class=\\"smb-gallery-btn-nav\\">
        <div>
        
        </div>
        <div>
          <a role=\\"smb-gallery-prev\\" class=\\"btn btn-primary disabled\\" href=\\"#\\"><i class=\\"fas fa-angle-left\\"></i> zur√ºck</a>
          <a role=\\"smb-gallery-next\\" class=\\"btn btn-primary \\" href=\\"#\\">weiter <i class=\\"fas fa-angle-right\\"></i></a>
        </div>
      </div>
    </div>
  "
`;
